src/resource.ts

```diff
@@ -206,13 +206,25 @@ export class Resource<T = any> extends EventEmitter {
    * Sends a PATCH request to the resource.
    *
    * This function defaults to a application/json content-type header.
+   *
+   * If the server responds with 200 Status code this will return a State object
+   *
+   * Otherwise it'll just clear the resource cache
    */
-  async patch(options: PatchRequestOptions): Promise<void> {
+  async patch(options: PatchRequestOptions): Promise<void | State<T>> {
 
-    await this.fetchOrThrow(
+    const response = await this.fetchOrThrow(
       optionsToRequestInit('PATCH', options)
     );
 
+    if (response.status === 200) {
+      const state = await this.client.getStateForResponse(this.uri, response);
+      this.updateCache(state);
+
+      return state;
+    }
+
+    this.clearCache();
```

Comment:
This is already handled centrally for all non-safe methods here: 

https://github.com/badgateway/ketting/blob/master/src/client.ts#L256

Reply:
Ahh, didn't notice that.

When is that run before or after this?